### ISSUE TITLE
Is it possible to raise an error/warning when trying to JSX render an undefined property of an object ?

### ISSUE THREAD

**Issue by cbdeveloper** [yannickcr/eslint-plugin-react#2734]
I have this pattern that I use a lot with `react` and `styled-components`

```
const LS = {};

LS.Title_DIV = styled.div`
  // TITLE CSS RULES
`;

// NOTE: I "FORGOT" TO DEFINE "LS.Subtitle_DIV"

function SomeComponent() {

  return(
    <React.Fragment>
      <LS.Title_DIV>Some Title</LS.Title_DIV>
      <LS.Subtitle_DIV>Some Subtitle</LS.Subtitle_DIV>     // <==== NOTE THAT LS.Subtitle_DIV is undefined
    </React.Fragment>
  );

}
```

The problem is that does not trigger any errors or warning. Not from Typescript, not from eslint, not from VSCode intellisense, I get nothing.

[![enter image description here][1]][1]

It even compiles successfully

[![enter image description here][2]][2]

It obviously does not run properly, but I get a generic runtime error message that usually takes me a while to figure out what is going on.

[![enter image description here][3]][3]

**QUESTION**

How can I set up Eslint so I get get an error or warning from that mistake on compile time? This happens quite often, but I really like this pattern, 'cause it allows me to quickly differentiate local styled-components from other components that where imported from other files.

**Note:** I'm not sure, but I thought that this plugin would be the place to get this error/warning, since it is responsible for the JSX linting, am I right?

Thanks.

  [1]: https://i.stack.imgur.com/58XOK.png
  [2]: https://i.stack.imgur.com/LE8CE.png
  [3]: https://i.stack.imgur.com/xsNAA.png

**Comment by ljharb** [comment]
It's not practical for this to be done statically. In the exact case you provided - a typo on an object literal defined in the same file - it could be done, but the vast majority of these cases will be from an imported object, or one returned by a function.

This is the realm of either unit tests, a type system, or both, not a linter.

**Comment by cbdeveloper** [comment]
@ljharb thanks for your reply!
